### PR TITLE
ZA: fix the astonishingly slow display of blog pages

### PR DESCRIPTION
**Pull request by mhl**
In combination, these commits take the time for the SQL on blog pages
down to 0.1 second from about 30 seconds.

### CODE REVIEW

File: pombola/south_africa/templates/info/_blog_sidebar.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
       {% endfor %}
     </ul>
 
-    {% if popular_posts.exists %}
+    {% if some_popular_posts %}
```
Comment (dracos) [inline pombola/south_africa/templates/info/_blog_sidebar.html]: DO you want to have an `{% empty %}` section here too, now that it's theoretically possible to get here but not display anything? (If all the blog posts are old, for example.)

Comment (dracos) [inline pombola/south_africa/templates/info/_blog_sidebar.html]: Or I guess just the date bit could be in the first query if that doesn't slow it down.

**Comment by dracos** [comment]
Wow, that was slow. Good work :)

File: pombola/south_africa/templates/info/_blog_sidebar.html (hunk 1 above)
Comment (mhl) [inline pombola/south_africa/templates/info/_blog_sidebar.html]: I've just added an empty block; I think it'll very rarely be seen so probably not a big deal anyway

**Pull request merged**